server/rollback_runner.go

```diff
@@ -0,0 +1,274 @@
+package server
+
+import (
+	"bufio"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"strings"
+	"sync"
+
+	"github.com/pingcap/tidb/parser"
+	"github.com/pingcap/tidb/parser/ast"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+
+	"github.com/bytebase/bytebase/api"
+	"github.com/bytebase/bytebase/common"
+	"github.com/bytebase/bytebase/common/log"
+	"github.com/bytebase/bytebase/plugin/db"
+	"github.com/bytebase/bytebase/plugin/db/mysql"
+	bbparser "github.com/bytebase/bytebase/plugin/parser"
+	"github.com/bytebase/bytebase/resources/mysqlutil"
+)
+
+var (
+	generateRollbackSQLChan = make(chan *api.Task, 100)
+)
+
+// NewRollbackRunner creates a new backup runner.
+func NewRollbackRunner(server *Server) *RollbackRunner {
+	return &RollbackRunner{
+		server: server,
+	}
+}
+
+// RollbackRunner is the backup runner scheduling automatic backups.
+type RollbackRunner struct {
+	server *Server
+}
+
+// Run is the runner for backup runner.
+func (r *RollbackRunner) Run(ctx context.Context, wg *sync.WaitGroup) {
+	defer wg.Done()
+	r.retryGetRollbackSQL(ctx)
+	for {
+		select {
+		case task := <-generateRollbackSQLChan:
+			if task.Instance.Engine == db.MySQL && task.Type == api.TaskDatabaseDataUpdate {
```

Comment:
Is this something that should be handled by the caller? Or what if this isn't true.